### DISCUSSION TITLE
How to save uploaded file into dynamic path

### DISCUSSION THREAD

**Discussion by massimomarazzi** [filamentphp/filament#5304] [Help] [answered]
Hi there, how to put files into dynamic path ?

i need to upload media over: ` logo/{MODEL_ID} OR {MODEL_NAME}/media.jpg`

How can i handle directory creation when i create a model?
With 

```
FileUpload::make('logo')
                    ->label('Logo')
                    ->preserveFilenames()
                    ->image()
                    ->maxSize(4096)
                   ->directory('logo/{DYNAMIC_DIR}'),
```

**Comment by ArtMin96** [comment, marked as answer]
This is not possible because the form is rendered when the record has not yet been created, so you don't have an id yet. You have to do it after submit.

**Reply by massimomarazzi** [reply]
Yes i know that. 
I need to know if there is a hook in filament starting after creating a model that allow to modify path or other prop.

Otherwise i can change path and move media inside event created .. but i think it is dirty solution.

**Reply by ArtMin96** [reply]
This is what you need [https://filamentphp.com/docs/2.x/admin/resources/creating-records#lifecycle-hooks](https://filamentphp.com/docs/2.x/admin/resources/creating-records#lifecycle-hooks)

**Reply by danharrin** [reply]
I would just use Spatie media library which handles it for you
